File: HW/HW1/JavaCode/Palindrome.java
package HW.HW1.JavaCode;


public class Palindrome{
    public static void main(String[] args) {
        Palindrome test = new Palindrome();
        String a = "Testtset";
        String b = "Test";
        System.out.println(test.isPalindrome(a));
        System.out.println(test.isPalindrome(b));
    }
    public boolean isPalindrome(String text){
        int start = 0;
        int last = text.length() - 1;
        while(start <= last) {
        	char currFirst = text.charAt(start);
        	char currLast = text.charAt(last);
        	if (!Character.isLetterOrDigit(currFirst)) {
        		start++;
        	} else if(!Character.isLetterOrDigit(currLast)) {
        		last--;
        	} else {
        		if (Character.toLowerCase(currFirst) != Character.toLowerCase(currLast)) {
        			return false;
        		}
        		start++;
        		last--;
        	}
        }
        return true;
    }
}

File: HW/HW1/PythonCode/Palindrome.py
class Palindrome:
    def isPalindrome(self, text):
        start = 0
        last = len(text) - 1
        while start <= last:
            currFirst = text[start]
            currLast = text[last]
            if not currFirst.isalnum():
                start += 1
            elif not currLast.isalnum():
                last -= 1
            else:
                if currFirst.lower() != currLast.lower():
                    return False
            start += 1
            last -= 1
        return True

test = Palindrome()
a = "Testtset"
b = "Test"
TestA = test.isPalindrome(a)
TestB = test.isPalindrome(b)
print(TestA)
print(TestB)